subcommand/server-acl-init/command_test.go

```diff
@@ -406,6 +407,63 @@ func TestRun_ConnectInjectAuthMethod(t *testing.T) {
 	}
 }
 
+// Test that ACL binding rules are updated if the rule selector changes.
+func TestRun_BindingRuleUpdates(t *testing.T) {
+	t.Parallel()
+	k8s, agent := completeSetup(t, resourcePrefix)
+	setUpK8sServiceAccount(t, k8s)
+	defer agent.Shutdown()
+	require := require.New(t)
+	consul := agent.Client()
+
+	ui := cli.NewMockUi()
+	firstRunArgs := []string{
+		"-server-label-selector=component=server,app=consul,release=" + releaseName,
+		"-resource-prefix=" + resourcePrefix,
+		"-k8s-namespace=" + ns,
+		"-expected-replicas=1",
+		"-create-inject-auth-method",
+		"-acl-binding-rule-selector=serviceaccount.name!=default",
+	}
+	// Our second run, we change the binding rule selector.
+	secondRunArgs := append(firstRunArgs,
+		"-create-inject-auth-method",
+		"-acl-binding-rule-selector=serviceaccount.name!=changed",
+	)
+
+	// Run the command first to populate the binding rule.
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	responseCode := cmd.Run(firstRunArgs)
+	require.Equal(0, responseCode, ui.ErrorWriter.String())
```

Comment:
This needs a check that the binding rule matched the first argument after the first run.

Reply:
In these update tests the pattern has been to only check that it changed. The first run creation is tested in other tests. It was done this way to reduce the amount of test code. If it breaks for the first run, it will be detected in the other tests. I've added it though.